libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -92,22 +92,29 @@ impl Greenlight {
         // Query for the existing credentials
         let mut parsed_credentials =
             Self::get_node_credentials(config.network, &signer, persister.clone())?
-                .ok_or(anyhow!("No credentials found"));
+                .ok_or(NodeError::Generic(anyhow!("No credentials found")));
         if parsed_credentials.is_err() {
             info!("No credentials found, trying to recover existing node");
             parsed_credentials = match Self::recover(config.network, seed.clone()).await {
                 Ok(creds) => Ok(creds),
```

Comment:
`recover` also works, so perhaps `restore_only` is a bit misleading?

Maybe `deny_register` or `deny_new_nodes` is better?

Reply:
`recover` only recovers the device cert/key from GL if the node exists, so it's a valid way to find out if a node exists for the seed.